File: FrontEnd/src/app/models/user.dto.ts
export interface UserDto {
  username: string;
  password: string;
  rol?: string;
}

export interface LoginResponse {
  token: string;
  user?: {
    id: number;
    username: string;
    rol: string;
  };
}

export enum UserRole {
  Cliente = 'Cliente',
  Administrador = 'Administrador',
  Empleado = 'Empleado'
}

File: BackEnd/Dtos/UserDto.cs
﻿using SistemaReservasApi.Enums;

namespace SistemaReservasApi.Dtos
{
    public class UserDto
    {
        public string Username { get; set; } = string.Empty;
        public string Password { get; set; } = string.Empty;
        public UserRole? Rol { get; set; }
    }
}